go-eicio/reader.go

```diff
@@ -3,46 +3,59 @@ package eicio
 import (
 	"encoding/binary"
 	"io"
+	"os"
 )
 
 type Reader struct {
 	byteReader io.Reader
 }
 
+func Open(filename string) (*Reader, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	return NewReader(file), nil
+}
+
+func (rdr *Reader) Close() {
+	rdr.byteReader.(*os.File).Close()
+}
+
 func NewReader(byteReader io.Reader) *Reader {
 	return &Reader{
 		byteReader: byteReader,
 	}
 }
 
-func (rdr *Reader) GetEvent() (event *Event, err error) {
+func (rdr *Reader) GetEvent() *Event {
```

Comment:
I was playing around with the idea (not very seriously) of leaving off the error in the return because I was thinking of possibly having the method deal with errors on its own, returning nil if the errors are unrecoverable.  However, it's true that the user needs to know that there is a difference between an abnormal unrecoverable error and an EOF for example.

To be more specific, I envision a magic number being added to the scheme to resynchronize with the stream if there is some hiccup.  In that case, what would you expect GetEvent() to return?